core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -233,6 +234,13 @@ public boolean onConnectionRequest(InetSocketAddress inetSocketAddress) {
         } else {
             ip = "<IP address withheld>";
         }
+
+        ConnectionRequestEvent requestEvent = new ConnectionRequestEvent(ip);
```

Comment:
Won't this fail if the ip is set to withheld above?

Reply:
Imo this may be more useful if both the original InetSocketAddress (keep as such) as well as the proxied ip was included, as this would allow DDoS mitigation providers to make plugins with dynamic lists that limit connections on the basis of the source IP.

Also please include a debug log for if the event is cancelled.